### PR TITLE
coll_hcoll: register progress callback directly without a proxy

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: ompi/mca/coll/hcoll/coll_hcoll_module.c [hunk 1]
```diff
@@ -290,7 +280,8 @@ mca_coll_hcoll_comm_query(struct ompi_communicator_t *comm, int *priority)
            mxm bcol in libhcoll needs world_group fully functional during init
            world_group, i.e. ompi_comm_world, is not ready at hcoll component open
            call */
-        opal_progress_register(mca_coll_hcoll_progress);
+        printf("hcoll_progress_fn=%p\n", hcoll_progress_fn);
```
Comment (bosilca) [inline ompi/mca/coll/hcoll/coll_hcoll_module.c]: Extraneous debug.

File: ompi/mca/coll/hcoll/coll_hcoll_module.c [hunk 2]
```diff
@@ -290,7 +280,8 @@ mca_coll_hcoll_comm_query(struct ompi_communicator_t *comm, int *priority)
...
+        opal_progress_register(hcoll_progress_fn);
```
Comment (bosilca) [inline ompi/mca/coll/hcoll/coll_hcoll_module.c]: The code will be more readable if the progress registration was the last step of a successful initialization.

Comment (bosilca) [reply ompi/mca/coll/hcoll/coll_hcoll_module.c]: In fact the progress function should only be registered when coll_module_enable is called.

Comment (yosefe) [reply ompi/mca/coll/hcoll/coll_hcoll_module.c]: @jladd-mlnx @vspetrov what do you think? is it possible to move or is progress needed for some phase during initialization?

Comment (jladd-mlnx) [reply ompi/mca/coll/hcoll/coll_hcoll_module.c]: @vspetrov do we need it early for non-blocking context create?

Comment (vspetrov) [reply ompi/mca/coll/hcoll/coll_hcoll_module.c]: No, i don't think we need hcoll internal progress to be registered into opal before we launch first collective via hcoll.

**Comment by yosefe** [comment]
removed the printf(), but i prefer to not move hcoll progress to module_enable now - will also need to add accounting of how many times the module was opened/closed/enabled, since it's done per communicator and progress register/unregister is global.

**Comment by jladd-mlnx** [comment]
👍

**Pull request merged**